groups/instance/cloud-init/templates/artifactory.yml.tpl

```diff
@@ -0,0 +1,26 @@
+#cloud-config
+fqdn: ${instance_fqdn}
+runcmd:
+  - wget https://dev.mysql.com/get/Downloads/Connector-J/mysql-connector-j-8.0.31.zip
+  - yum install unzip -y
+  - yum install mariadb -y
+  - unzip mysql-connector-j-8.0.31.zip
+  - cp mysql-connector-j-8.0.31/mysql-connector-j-8.0.31.jar ~/artifactory/var/bootstrap/artifactory/tomcat/lib
+  - mysql --version
+  - mysql -h ${db_fqdn} -P ${db_port} -u ${db_username} -p ${db_password} -e "CREATE DATABASE artdb CHARACTER SET utf8 COLLATE utf8_bin;"
+  - mysql -h ${db_fqdn} -P ${db_port} -u ${db_username} -p ${db_password} -e "CREATE USER '${db_username}'@'%' IDENTIFIED BY '${db_password}';"
+  - mysql -h ${db_fqdn} -P ${db_port} -u ${db_username} -p ${db_password} -e "GRANT ALL on artdb.* TO '${db_username}'@'%';"
+  - mysql -h ${db_fqdn} -P ${db_port} -u ${db_username} -p ${db_password} -e "FLUSH PRIVILEGES;"
+  - mysql -h ${db_fqdn} -P ${db_port} -u ${db_username} -p ${db_password} -e "SHOW DATABASES;"
+  - service artifactory enable
+  - service artifactory start
+write-files:
+  - path: ~/artifactory/var/etc
```

Comment:
The use of `~` in the paths here will evaluate to home directory of the user cloud-init runs as (root). You should use an absolute path to ensure the files are created in the correct directory as required by Artifactory

Also, are the `mysql` commands idempotent? Will they work/fail if the instance needs to be re-deployed, such as for an application update?

Reply:
Would be grateful for suggestions on how to modify [MySQL setup](https://www.jfrog.com/confluence/display/JFROG/MySQL).

Reply:
SQL commands removed